docs/core/deploying/single-file.md

```diff
@@ -11,10 +11,32 @@ Bundling all application-dependent files into a single binary provides an applic
 
-There are some caveats that you need to be aware for single-file use, first of which is the use of path information to locate a file relative to the location of your application. The <xref:System.Reflection.Assembly.Location?displayProperty=nameWithType> API will return an empty string, which is the default behavior for assemblies loaded from memory. The compiler will give a warning for this API during build time to alert the developer to the specific behavior. If the path to the application directory is needed, the <xref:System.AppContext.BaseDirectory?displayProperty=nameWithType> API will return the directory where the AppHost (the single-file bundle itself) resides. Managed C++ applications aren't well suited for single-file deployment and we recommend that you write applications in C# to be single-file compatible.
+## API incompatibility
+
+There are some caveats that you need to be aware for single-file use, first of which is the use of path information to locate a file relative to the location of your application. The <xref:System.Reflection.Assembly.Location?displayProperty=nameWithType> API will return an empty string, which is the default behavior for assemblies loaded from memory. The compiler will give a [warning](https://docs.microsoft.com/visualstudio/code-quality/il3000) for this API during build time to alert the developer to the specific behavior. If the path to the application directory is needed, the <xref:System.AppContext.BaseDirectory?displayProperty=nameWithType> API will return the directory where the application executable resides. There is also an option, `ExcludeFromSingleFile`, to exclude files from being bundled that is described [below](#exclude-files-from-being-embedded) in detail.
+
+The table below has the relevant runtime library API details for single-file use.
+
+| API                                                                                     | Note |
+| ---------------------------------------------------------------------------------------  | ------- |
+| `Assembly.Location` | Returns an empty string, use `AppContext.BaseDirectory` |
+| `Module.FullyQualifiedName`  | Returns a `<Unknown>` string |
+| `Module.Name`  | Returns a `<Unknown>` string |
+| `Assembly.GetFile`  | Will throw `IOException` |
+| `Assembly.GetFiles`  | Will throw `IOException` |
+| `Assembly.CodeBase`  | Will throw `PlatformNotSupportedException` |
+| `Assembly.EscapedCodeBase`  | Will throw `PlatformNotSupportedException` |
+| `AssemblyName.CodeBase`  | Returns null |
+| `AssemblyName.EscapedCodeBase`  | Returns null |
+
```

Comment:
```suggestion

We have some recommendations for fixing common scenarios:

* To access files next to the executable, use <xref:System.AppContext.BaseDirectory?displayProperty=nameWithType>

* To find the file name of the executable, use the first element of <xref:System.Environment.GetCommandLineArgs()?displayProperty=nameWithType>

* To avoid shipping loose files entirely, consider using [embedded resources](https://docs.microsoft.com/en-us/dotnet/framework/resources/creating-resource-files-for-desktop-apps)
```